modules/chat/src/js/chat.js

```diff
@@ -146,6 +154,57 @@ const chat = function(){
   };
 
 
+  // purpose:		loads messages from given page
+  // arguments:	the page number (int, default: 1)
+  //            items per page to get (int, default: 30)
+  // ------------------------------------------------------------------------
+  module.loadPage = (page = 1, perPage = 30) => {
+    let latestMessage = module.settings.messagesList.querySelector('li:first-child');
+
+    // show the loading indicator at start
+    module.settings.loadingIndicator.style.display = 'block';
+
+    // get the data
+    fetch(`/api/chat/get_messages?conversation_id=${module.conversationId}&page=${page}&per_page=${perPage}`)
+    .then(response => {
+      // parse it to JSON if valid
+      if(response.ok){
+        return response.json();
+      } else {
+        return Promise.reject(response);
+      }
+    })
+    .then((data) => {
+      // construct all the HTML elements for messages
+      let html = '';
+
+      Object.entries(data.results).reverse().forEach(([key, data]) => {
+        data = Object.assign(data, { status: (module.settings.currentUserId == data.from_id) ? 'sent' : 'received'});
+
+        html += module.settings.messageTemplate(data);
+      });
+
+      // put the messages on top
+      module.settings.messagesList.insertAdjacentHTML('afterbegin', html);
+
+      // disable loading next pages if there is nothing left
+      if(!data.has_next_page){
+        module.settings.morePages = false;
+      }
+    })
+    .catch((error) => {
+      console.log(error);
+      error.json().then(data => console.log(data));
+    })
+    .finally(() => {
+      // remove the loading indicator
```

Comment:
I don't like idea of adding comment to every single line.

Reply:
Every new person who will try to use this as an example of building the chat will. ;)

Reply:
Me especially. I'll be doing a web sockets instant messaging video tutorial This will help me greatly.